### PR TITLE
Support for BPMN timer events

### PR DESCRIPTION
**Pull request by CTI777**
Related to https://github.com/Activiti/Activiti/issues/2636
Depends on 
https://github.com/Activiti/activiti-api/pull/121
https://github.com/Activiti/Activiti/pull/2753
https://github.com/Activiti/activiti-cloud-api/pull/97
https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/322

### CODE REVIEW

File: activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/TimerCanceledEventConverter.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import org.activiti.api.process.model.events.BPMNTimerEvent;
+import org.activiti.cloud.api.model.shared.events.CloudRuntimeEvent;
+import org.activiti.cloud.api.model.shared.impl.events.CloudRuntimeEventImpl;
+import org.activiti.cloud.api.process.model.events.CloudBPMNTimerCanceledEvent;
+import org.activiti.cloud.api.process.model.impl.events.CloudBPMNTimerCanceledEventImpl;
+import org.activiti.cloud.services.audit.jpa.events.AuditEventEntity;
+import org.activiti.cloud.services.audit.jpa.events.TimerCanceledAuditEventEntity;
+
+public class TimerCanceledEventConverter extends BaseEventToEntityConverter {
```
Comment (ffazzini) [inline activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/TimerCanceledEventConverter.java]: same suggestion from previous prs about generics might apply here as well?

File: activiti-cloud-services-audit-jpa/src/test/java/org/activiti/cloud/services/audit/jpa/controller/AuditEventsControllerImplIT.java [hunk 2]
```diff
@@ -345,4 +349,44 @@ public void getSignalEventById() throws Exception {
                 .andDo(print())
                 .andExpect(status().isOk());
     }
+    
+    @Test
+    public void getTimerEventById() throws Exception {
```
Comment (erdemedeiros) [inline activiti-cloud-services-audit-jpa/src/test/java/org/activiti/cloud/services/audit/jpa/controller/AuditEventsControllerImplIT.java]: Could you rename this test to `shouldGetTimerEventById`?

File: activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/TimerCancelledEventConverter.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Override
+    public String getSupportedEvent() {
+        return BPMNTimerEvent.TimerEvents.TIMER_CANCELLED.name();
+    }
+
+    @Override
+    protected TimerCancelledAuditEventEntity createEventEntity(CloudRuntimeEvent cloudRuntimeEvent) {
+        CloudBPMNTimerCancelledEvent cloudEvent = (CloudBPMNTimerCancelledEvent) cloudRuntimeEvent;
+        return new TimerCancelledAuditEventEntity(cloudEvent.getId(),
```
Comment (eromano) [inline activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/TimerCancelledEventConverter.java]: Uncle Bob in the Clean Code book explains that an ideal function is a niladic function (0 arguments)
1 argument monadic function Not ideal
2 arguments dyadic function Not ideal
3 arguments triadic function
More then  3 arguments polyadic function No to be used

Would not be better define an interface as an input parameter and let all the object implement that interface, and do the assignment job only in the super object?

**Review by eromano — CHANGES_REQUESTED** [review]

File: activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/TimerCancelledEventConverter.java (hunk 3 above)
Comment (erdemedeiros) [reply activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/TimerCancelledEventConverter.java]: Totally agree with Eugenio. All the events are using big constructors, but we need to stop doing so.

**Review by erdemedeiros — CHANGES_REQUESTED** [review]

File: activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/TimerCancelledEventConverter.java (hunk 3 above)
Comment (CTI777) [reply activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/TimerCancelledEventConverter.java]: I do not like big constructors as well... But I have tried to make everything close / similar to already implemented EventEntities... It means that we have to change all, in principle... Should I apply changes for all eventEntities here, or better to make a new PR and apply changes in one shot?

Comment (erdemedeiros) [reply activiti-cloud-services-audit-jpa/src/main/java/org/activiti/cloud/services/audit/jpa/converters/TimerCancelledEventConverter.java]: Now we fixed that for timers, but you're right, we need to fix that for other events too. Let's do that in a separated PR.

**Review by erdemedeiros — APPROVED** [review]

**Pull request merged**